packages/compass-assistant/src/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { registerCompassPlugin } from '@mongodb-js/compass-app-registry';
+import { AssistantProvider } from './assistant-provider';
+import { Chat } from './@ai-sdk/react/chat-react';
+import { DocsProviderTransport } from './docs-provider-transport';
+import { atlasServiceLocator } from '@mongodb-js/atlas-service/provider';
+
+const CompassAssistantProvider = registerCompassPlugin(
+  {
+    name: 'CompassAssistant',
+    component: AssistantProvider,
+    activate: (initialProps, { atlasService }) => {
+      // TODO: We will temporarily default to the staging url until the docs
+      // API is deployed to the production environment.
+      const baseUrl = process.env.COMPASS_ASSISTANT_USE_ATLAS_SERVICE_URL
+        ? atlasService.assistantApiEndpoint()
+        : 'https://knowledge.staging.corp.mongodb.com/api/v1';
+      const chat = new Chat({
+        transport: new DocsProviderTransport({
```

Comment:
@gribnoysup moved the transport inside the activate, I was originally exposing it for mocking (which I didn't end up doing) but I don't need to mock at such a low-level anyhow.. Not sure if the defaulting to staging bit for now is fine, just figured it'd be good to make this simple with the feature flag.